File: iPhone/Apex Cougars/Classes/TeamRosterService.m
//
//  TeamRosterService.m
//  iAthletics Framework
//
//  Created by Robert Signore on 6/17/10.
//  Copyright 2010 Appapro. All rights reserved.
//

#import "TeamRosterService.h"


@implementation TeamRosterService

@synthesize athletes;
@synthesize currentAthlete;

#pragma mark Constructors/Destructors

-(id) initWithApplication:(iAthleticsApplication *)theApp
{
	self = [super initWithApplication:theApp];
	
	NSMutableArray *newArray = [[NSMutableArray alloc]initWithCapacity:100];
	self.athletes = newArray;
	[newArray release];
	
	return self;
}

-(void)dealloc
{
	[self.athletes release];
	[self.currentAthlete release];
	
	[super dealloc];
}

#pragma mark service request methods

-(NSArray *)requestServiceWithTeamId:(NSInteger)teamId
{
	// create the parameter string
	//
	NSString *paramString = [[NSString alloc]initWithFormat:@"teamId=%d", teamId];
	[self requestServiceName:@"GetTeamRoster" parameters:paramString];
	
	[paramString release];
	
	
	return self.athletes;		
}

#pragma mark -
#pragma mark iAthleticsService ABSTRACT methods

-(void) elementStart:(NSString *)elementName
{
	// prepare to receive new AppSeason Info
	//
	if([elementName compare:@"Player"] == NSOrderedSame)
	{
		self.currentAthlete = nil;
		
		Athlete *newAthlete = [[Athlete alloc]init];
		self.currentAthlete = newAthlete;
		[newAthlete release];
	}
}

-(void)elementEnd:(NSString *)elementName withText:(NSString *)text
{
	
	// player name
	//
	if([elementName compare:@"Name"] == NSOrderedSame)
	{
		self.currentAthlete.name = text;
	}
	
	// player number
	//
	else if([elementName compare:@"Number"] == NSOrderedSame)
	{
		self.currentAthlete.number = text;
	}	
	
	// grade
	//
	else if([elementName compare:@"Grade"] == NSOrderedSame)
	{
		self.currentAthlete.grade = text;
	}	
	
	// height
	//
	else if([elementName compare:@"Height"] == NSOrderedSame)
	{
		self.currentAthlete.height = text;
	}
	
	// weight
	//
	else if([elementName compare:@"Weight"] == NSOrderedSame)
	{
		self.currentAthlete.weight = text;
	}
	
	// positions
	//
	else if([elementName compare:@"Positions"] == NSOrderedSame)
	{
		self.currentAthlete.positions = text;
	}
	
	// if this is the end of the NewsLink then add the currentLink to the 
	// links array
	else if([elementName compare:@"Player"] == NSOrderedSame)
	{
		[self.athletes addObject:self.currentAthlete];
	}
}

@end

File: Android/iAthletics/src/com/appapro/iathletics/services/TeamRosterService.java
package com.appapro.iathletics.services;

import java.util.ArrayList;
import java.util.Collections;

import com.appapro.iathletics.activities.iAthleticsActivity;
import com.appapro.iathletics.types.Athlete;

public class TeamRosterService extends iAthleticsService
{
	ArrayList<Athlete> athletes = new ArrayList<Athlete>(10);
	Athlete currentAthlete;
	
	public TeamRosterService(iAthleticsActivity activity)
	{
		super(activity);
	}
	
	

	public ArrayList<Athlete> getAthleties(int teamId)
	{
		String parameters = "teamId=" + String.valueOf(teamId);
		
		this.requestService("GetTeamRoster", parameters);
		
		Collections.sort(athletes);
		return athletes;
	}


	@Override
	protected void elementStart(String elementName)
	{
		// prepare to receive new athlete Info
		//
		if(elementName.compareTo("Player") == 0)
		{
			this.currentAthlete = new Athlete();
		}
	}

	@Override
	protected void elementEnd(String elementName, String text)
	{
		// player name
		//
		if(elementName.compareTo("Name") == 0)
		{
			this.currentAthlete.setName(text);
		}
		
		// player number
		//
		else if(elementName.compareTo("Number") == 0)
		{
			this.currentAthlete.setNumber(text);
		}	
		
		// grade
		//
		else if(elementName.compareTo("Grade") == 0)
		{
			this.currentAthlete.setGrade(text);
		}	
		
		// height
		//
		else if(elementName.compareTo("Height") == 0)
		{
			this.currentAthlete.setHeight(text);
		}
		
		// weight
		//
		else if(elementName.compareTo("Weight") == 0)
		{
			this.currentAthlete.setWeight(text);
		}
		
		// positions
		//
		else if(elementName.compareTo("Positions") == 0)
		{
			this.currentAthlete.setPositions(text);
		}
		
		// if this is the end of the Player then add the currentLink to the 
		// links array
		else if(elementName.compareTo("Player") == 0)
		{
			this.athletes.add(currentAthlete);
		}
	}
}
